hail/python/hail/expr/functions.py

```diff
@@ -3736,20 +3755,23 @@ def deep_flatten(l):
         return result
 
     if isinstance(collection, np.ndarray):
-        nd = collection.astype(np.float64)
-        data = list(nd.flat)
-        shape = nd.shape
-    elif isinstance(collection, list):
+        if row_major is None:
+            return to_expr(collection)
+        else:
+            elem_type = np_type_to_hl_type(collection.dtype)
+            data = [to_expr(i.item(), elem_type) for i in collection.flatten('C' if row_major else 'F')]
+            shape = collection.shape
+    else:
+        if row_major is None:
```

Comment:
`assert isinstance(collection, list)`

Reply:
It's not necessarily a list, it could be a scalar value. But I can make that more explicit.